src/Magento/FunctionalTestingFramework/DataGenerator/Persist/Curl/AdminExecutor.php

```diff
@@ -158,7 +159,9 @@ public function read($successRegex = null, $returnRegex = null)
         if (!empty($returnRegex)) {
             preg_match($returnRegex, $this->response, $returnMatches);
             if (!empty($returnMatches)) {
-                return $returnMatches;
+                return isset($returnIndex, $returnMatches[$returnIndex])
+                    ? $returnMatches[$returnIndex]
+                    : $returnMatches;
```

Comment:
@jilu1 
- So in case **returnIndex** is not defined we will return full pattern match string instead of array?
- Should we throw an error in case **returnIndex** is specified but not present in **$returnMatches** array?

Reply:
Fixed